### PR TITLE
WT-4480 Fix duplicate backup cursor and archiving check.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

File: src/cursor/cur_backup.c [hunk 1]
```diff
@@ -239,15 +241,15 @@ __backup_start(WT_SESSION_IMPL *session,
 	 * Single thread hot backups: we're holding the schema lock, so we
 	 * know we'll serialize with other attempts to start a hot backup.
 	 */
-	if (conn->hot_backup && other == NULL)
+	if (conn->hot_backup && !is_dup)
 		WT_RET_MSG(
 		    session, EINVAL, "there is already a backup cursor open");
 
 	if (F_ISSET(session, WT_SESSION_BACKUP_DUP) && other != NULL)
```
Comment (keithbostic) [inline src/cursor/cur_backup.c]: Why are testing `other != NULL` explicitly?

File: src/cursor/cur_backup.c [hunk 2]
```diff
@@ -223,11 +224,12 @@ __backup_start(WT_SESSION_IMPL *session,
 	WT_DECL_RET;
 	WT_FSTREAM *srcfs;
 	const char *dest;
-	bool exist, log_only, target_list;
+	bool exist, is_dup, log_only, target_list;
 
 	conn = S2C(session);
 	srcfs = NULL;
 	dest = NULL;
+	is_dup = other != NULL;
```
Comment (keithbostic) [inline src/cursor/cur_backup.c]: As far as I can tell, we're not actually using `other` for any purpose in this function, why not pass `other != NULL` to `__backup_start()`, as a boolean `is_dup`?

Comment (sueloverso) [reply src/cursor/cur_backup.c]: Good idea. Fixed.

**Comment by sueloverso** [comment]
autobuild: retest this please

**Comment by sueloverso** [comment]
autobuild: retest this please

**Pull request merged**